modules/auth/oauth2/oauth2.go

```diff
@@ -0,0 +1,64 @@
+package oauth2
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/log"
+	"github.com/gorilla/sessions"
+	"github.com/markbates/goth"
+	"github.com/markbates/goth/gothic"
+	"github.com/markbates/goth/providers/github"
+	"net/http"
+	"os"
+	"github.com/satori/go.uuid"
+	"path/filepath"
+)
+
+var (
+	sessionUsersStoreKey = "gitea-oauth2-sessions"
+	providerHeaderKey    = "gitea-oauth2-provider"
+)
+
+func init() {
+	sessionDir := filepath.Join(setting.AppDataPath, "sessions", "oauth2")
```

Comment:
Was already creating such a function for registering the activated OAuth providers 😄 
Will add this as well